src/Nethermind/Nethermind.History.Test/HistoryPrunerTests.cs

```diff
@@ -0,0 +1,336 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Nethermind.Blockchain.Receipts;
+using Nethermind.Config;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Specs;
+using Nethermind.Core.Test;
+using Nethermind.Core.Test.Blockchain;
+using Nethermind.Logging;
+using Nethermind.Specs;
+using Nethermind.State.Repositories;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace Nethermind.History.Test;
+
+public class HistoryPrunerTests
+{
+    private const long SecondsPerSlot = 1;
+    private const long BeaconGenesisBlockNumber = 50;
+    private static readonly ulong BeaconGenesisTimestamp = (ulong)new DateTimeOffset(TestBlockchain.InitialTimestamp).ToUnixTimeSeconds() + (BeaconGenesisBlockNumber * SecondsPerSlot);
+
+    [Test]
+    public async Task Can_prune_blocks_older_than_specified_epochs()
+    {
+        const int Blocks = 100;
+
+        using BasicTestBlockchain testBlockchain = await BasicTestBlockchain.Create();
+
+        List<Hash256> blockHashes = [];
+        blockHashes.Add(testBlockchain.BlockTree.Head!.Hash!);
+        for (int i = 0; i < Blocks; i++)
+        {
+            await testBlockchain.AddBlock();
+            blockHashes.Add(testBlockchain.BlockTree.Head!.Hash!);
+        }
+
+        Core.Block head = testBlockchain.BlockTree.Head;
+        Assert.That(head, Is.Not.Null);
+
+        // n.b. technically invalid, should be at least 82125 epochs
+        // however not feasible to test this
+        IHistoryConfig historyConfig = new HistoryConfig
+        {
+            HistoryRetentionEpochs = 2,
+            DropPreMerge = false
+        };
+        ISpecProvider specProvider = Substitute.For<ISpecProvider>();
+
+        HistoryPruner historyPruner = new(
```

Comment:
Note: You can `testBlockchain.Container.Resolve<IHistoryPruner>()`. 

To change the config,  the `BasicTestBlockchain.Create()` or some overload, take a configurator func in which you can set the history config with sometthing like `.AddSingleton<IHistoryConfig>(theHistoryConfig)`.